targets/csharp/source/source/Uunit/PlayFabApiTest.cs

```diff
@@ -262,6 +264,31 @@ private void UserDataApiContinued3(PlayFabResult<GetUserDataResult> getDataResul
             testContext.True(testMin <= timeUpdated && timeUpdated <= testMax);
         }
 
+		
+		
+		/// <summary>
+        /// CLIENT API
+        /// Test several parallel requests and ensures they complete with no errors.
+        /// </summary>
+        [UUnitTest]
+        public void ParallelRequests(UUnitTestContext testContext)
+        {
+            var tasks = Enumerable.Range(0, 10)
+                .Select(_ => PlayFabClientAPI.GetUserDataAsync(new GetUserDataRequest(), _, extraHeaders).ThrowIfApiError());
+
+            Task.WhenAll(tasks).ContinueWith(token =>
+            {
+                if (!token.IsCanceled && !token.IsFaulted)
```

Comment:
This is a good debug test, but not a good long term test.
If 9 threads fail, and the last one passes, the state that's recorded will be "passed".
If 5 threads pass, and then 1 fails, and then 4 pass, the state that's recorded will be "passed".
In reality, you should only call "EndTest" once, with the correct state. You're calling "EndTest" ten times.
You'll need to figure out a way to perform this test such that the 10 threads run in parallel, you save the state of each somewhere, and then evaluate all states at the far end, and call EndTest just once.

Reply:
I'm not sure what you mean. Continuation runs on "Task.WhenAll" which is only triggered once all tasks complete. WhenAll task is faulted if one of children if faulted.

Reply:
IC, I misunderstood.